### ISSUE TITLE
KeyFor: overriding Collection.toArray(T[]) reports a spurious second override.typaram.invalid

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#1972]
Overriding `Collection.toArray(T[])` under the KeyFor checker reports a spurious second `override.typaram.invalid`, on top of whatever `override.return.invalid`/`override.param.invalid` diagnostic the pre-existing per-occurrence checks already gave:

```java
abstract static class MyColl<E> extends AbstractCollection<E> {
    @Override
    public <T> T[] toArray(T[] a) { return super.toArray(a); }   // new override.typaram.invalid
}
```

The root cause is a defaulting inconsistency in the KeyFor subchecker: a source-declared, unbounded `<T>`'s lower bound defaults to `@UnknownKeyFor NullType` (top), while the annotated JDK's own declaration of `Collection.toArray`, `<T> T[] toArray(@PolyNull T[] a)`, defaults `T`'s lower bound to `@KeyForBottom NullType`. Since #1961's `isTypeParameterBoundOverrideValid` check (added in #1961) compares the overriding method's own declared bound against the overridden one's, this pre-existing default mismatch -- which already independently produced an `override.return.invalid`/`override.param.invalid` diagnostic before #1961 -- now also produces a second, redundant `override.typaram.invalid`.

Overriding `toArray` is a common pattern, so this is real noise for anyone implementing a `Collection` under the KeyFor checker. Two possible fixes, either independent of the other:

- Fix the KeyFor lower-bound default itself -- a lower bound of *top* for an unbounded, unannotated type parameter looks like an outright bug (it should presumably default to bottom, like other checkers' unbounded type parameters do).
- Suppress `override.typaram.invalid` for a type parameter that a parameter/return occurrence check already independently rejected. This is the general form of #1973.

Found during #1961's review; not caused by #1961, only newly visible through it (the pre-existing `override.return.invalid`/`override.param.invalid` diagnostic already fired for this case; #1961 adds a second one on top).
